commit 58110c44429c711beeb872d918b64cf0e90d1de5
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Mon Feb 7 13:48:20 2011 +0100

    entropy: add forward compatibility with newer Entropy (exposing Client.PackageFileUpdates())

diff --git a/backends/entropy/entropyBackend.py b/backends/entropy/entropyBackend.py
index 59b148e..85e97c3 100755
--- a/backends/entropy/entropyBackend.py
+++ b/backends/entropy/entropyBackend.py
@@ -833,8 +833,12 @@ class PackageKitEntropyBackend(PackageKitBaseBackend, PackageKitEntropyMixin):
         self._log_message(__name__, "_generic_message:", decolorize(message))
 
     def _config_files_message(self):
-        scandata = self._entropy.FileUpdates.scan(dcache = True,
-            quiet = True)
+        if hasattr(self._entropy, "PackageFileUpdates"):
+            scandata = self._entropy.PackageFileUpdates().scan(dcache = True,
+                quiet = True)
+        else:
+            scandata = self._entropy.FileUpdates.scan(dcache = True,
+                quiet = True)
         if scandata is None:
             return
         if len(scandata) > 0:
